### PR TITLE
Add preview product button on products edit page

### PR DESCRIPTION
**Pull request by dimon-k**
The button is available under: `/admin/products/product-slug/edit`, right next to `+ New Product ` button:

preview product button

### CODE REVIEW

File: core/app/helpers/spree/products_helper.rb [hunk 1]
```diff
@@ -70,6 +70,10 @@ def available_status(product) # will return a human readable string
       end
     end
 
+    def frontend_available?
```
Comment (damianlegawiec) [inline core/app/helpers/spree/products_helper.rb]: @dimon-k let's move it to `BaseHelper`

File: backend/app/views/spree/admin/products/edit.html.erb [hunk 2]
```diff
@@ -1,4 +1,7 @@
 <% content_for :page_actions do %>
+  <% if frontend_available? %>
```
Comment (damianlegawiec) [inline backend/app/views/spree/admin/products/edit.html.erb]: 👍

**Review by damianlegawiec — CHANGES_REQUESTED** [review]

File: backend/spec/features/admin/products/edit/products_spec.rb [hunk 3]
```diff
@@ -24,17 +26,16 @@
...
-      within('table.table tbody tr:nth-child(1)') do
-        click_icon(:edit)
-      end
-
       fill_in 'product_slug', with: 'random-slug-value'
       click_button 'Update'
       expect(page).to have_content('successfully updated!')
     end
+
+    it 'has a link to preview product' do
```
Comment (damianlegawiec) [inline backend/spec/features/admin/products/edit/products_spec.rb]: `has a link to preview product` -> `has a link to preview a product`

File: backend/spec/features/admin/products/edit/products_spec.rb [hunk 4]
```diff
@@ -24,17 +26,16 @@
...
+      allow(Spree::Core::Engine).to receive(:frontend_available?).and_return(true)
+      allow_any_instance_of(Spree::BaseHelper).to receive(:product_url).and_return('#')
+      click_link 'Details'
+      page.find('#admin_preview_product')
```
Comment (bbonislawski) [inline backend/spec/features/admin/products/edit/products_spec.rb]: `expect(page).to have_css('#admin_preview_product')` please

**Review by bbonislawski — CHANGES_REQUESTED** [review]

File: backend/spec/features/admin/products/edit/products_spec.rb [hunk 5]
```diff
@@ -24,17 +26,16 @@
...
-      end
-
       fill_in 'product_slug', with: 'random-slug-value'
       click_button 'Update'
       expect(page).to have_content('successfully updated!')
     end
+
+    it 'has a link to preview product' do
+      allow(Spree::Core::Engine).to receive(:frontend_available?).and_return(true)
+      allow_any_instance_of(Spree::BaseHelper).to receive(:product_url).and_return('#')
```
Comment (damianlegawiec) [inline backend/spec/features/admin/products/edit/products_spec.rb]: `and_return('#')` -> `and_return('http://example.com/products/product-slug')`

**Review by damianlegawiec — CHANGES_REQUESTED** [review]

File: backend/spec/features/admin/products/edit/products_spec.rb [hunk 6]
```diff
@@ -24,17 +26,16 @@
...
       fill_in 'product_slug', with: 'random-slug-value'
       click_button 'Update'
       expect(page).to have_content('successfully updated!')
     end
+
+    it 'has a link to preview a product' do
+      allow(Spree::Core::Engine).to receive(:frontend_available?).and_return(true)
+      allow_any_instance_of(Spree::BaseHelper).to receive(:product_url).and_return('http://example.com/products/product-slug')
+      click_link 'Details'
+      expect(page).to have_css('#admin_preview_product')
```
Comment (damianlegawiec) [inline backend/spec/features/admin/products/edit/products_spec.rb]: @dimon-k sorry forgot, let's add a 

```
expect(page).to have_content('http://example.com/products/product-slug')
```

**Review by damianlegawiec — CHANGES_REQUESTED** [review]

File: backend/spec/features/admin/products/edit/products_spec.rb (hunk 6 above)
Comment (dimon-k) [reply backend/spec/features/admin/products/edit/products_spec.rb]: good point @damianlegawiec

**Review by damianlegawiec — COMMENTED** [review]
👍

**Review by damianlegawiec — APPROVED** [review]

**Pull request closed without merging**